packages/std/src/ado_contract/permissioning.rs

```diff
@@ -345,6 +347,48 @@ impl<'a> ADOContract<'a> {
             .collect::<Vec<String>>();
         Ok(actions)
     }
+
+    pub fn query_permissioned_actors(
+        &self,
+        deps: Deps,
+        action: impl Into<String>,
+        start_after: Option<u32>,
+        limit: Option<u32>,
+        order_by: Option<OrderBy>,
+    ) -> Result<Vec<String>, ContractError> {
+        let action_string: String = action.into();
+
+        let mut actors = permissions()
+            .keys(deps.storage, None, None, Order::Ascending)
+            .filter(|item| item.as_ref().unwrap().starts_with(&action_string))
+            .map(|item| {
+                let actor: String = item.unwrap_or_default()[action_string.len()..].to_string();
+                actor
+            })
+            .collect::<Vec<String>>();
+
+        let start = start_after.unwrap_or(0) as usize;
+        let limit = limit.unwrap_or(DEFAULT_QUERY_LIMIT).min(MAX_QUERY_LIMIT) as usize;
+
+        let (start, end) = match order_by {
+            Some(OrderBy::Desc) => (
+                actors
+                    .len()
+                    .saturating_sub(cmp::min(actors.len(), start + limit)),
+                actors.len().saturating_sub(cmp::min(start, actors.len())),
+            ),
+            // Default ordering is Ascending.
+            _ => (
+                cmp::min(actors.len(), start),
+                cmp::min(start + limit, actors.len()),
+            ),
+        };
+        let slice = &mut actors[start..end];
+        if order_by == Some(OrderBy::Desc) {
+            slice.reverse();
```

Comment:
I'm not sure this works. It'll take an ascending ordered list, slice off the first n elements and then reverse them, rather than reversing the list before slicing?